pennylane/qchem/factorization.py

```diff
@@ -226,28 +362,145 @@ def _double_factorization_cholesky(two, tol_factor=1.0e-10, shape=None, interfac
+    matrices :math:`U` (leaf tensors) and symmetric matrices :math:`Z` (core tensors) such that:
+
+    .. math::
+
+        V_{ijkl} \approx \sum_r^R \sum_{pq} U_{ip}^{(r)} U_{jp}^{(r)} Z_{pq}^{(r)} U_{kq}^{(r)} U_{lq}^{(r)}.
+
+    This is done by optimizing the following cost function :math:`\mathcal{L}` via a greedy
+    approach, i.e., we optimize the leaf-tensor pairs layer-by-layer (or one-by-one) instead
+    of optimizing them all at once as the latter has gives unfavorable performance (and scaling):
+
+    .. math::
+
+       \mathcal{L}(U, Z) = \frac{1}{2} \bigg|V_{ijkl} - \sum_r^R \sum_{pq} U_{ip}^{(r)} U_{jp}^{(r)} Z_{pq}^{(r)} U_{kq}^{(r)} U_{lq}^{(r)}\bigg|_{\text{F}} + \rho \sum_r^R \sum_{pq} \bigg|Z_{pq}^{(r)}\bigg|^{\gamma}.
+
+    First, leaf tensors :math:`U^{(r)} = \exp{(X^{(r)})}` are parameterized by the
+    anti-symmetric orbital rotations :math:`X^{(r)}` to compute the Frobenius norm
+    of the difference between the :math:`V` and the above approximation. A further
+    regularization term penalizing large terms in :math:`|Z^{(r)}|` is added to
+    this after scaling it with a constant factor :math:`\rho` to compute the loss.
+
+    Args:
+        two (array[array[float]]): two-electron integral tensor in the molecular orbital
+            basis arranged in chemist notation.
+        optimizer (optax.optimizer): an optax optimizer instance. If not provided, `Adam
+            <https://optax.readthedocs.io/en/latest/api/optimizers.html#optax.adam>`_ is
+            used with ``0.001`` learning rate.
+        num_factors (int): maximum number of factors that should be optimized for compressed
+            double factorization. Default is :math:`2\times N`, where `N` is the number of
+            dimensions of two-electron tensor.
+        num_steps (int): maximum number of epochs for optimizing each factor. Default is ``1000``.
+        init_params (dict[str, TensorLike] | None): intial values of the orbital rotations
+            (:math:`X`) and core tensors (:math:`Z`) of shape ``(num_factors, N, N)`` given as
+            a dictionary with keys ``"X"`` and ``"Z"``, where `N` is the number of dimension of
+            two-electron tensor. If not given, by default, zero matrices will be used.
+        prefactor (float): prefactor for scaling the regularization term. Default is ``1e-5``.
+        norm_order (int): type of regularization (``0``: None, ``1``: L1, and ``2``: L2) used
+            for optimizing. Default is to not include any regularization term.
+
+    Returns:
+        tuple(TensorLike, TensorLike): tuple containing core tensors and leaf tensors
+        approximating the two-electron integral tensor.
+    """
+    norb = two.shape[0]
+    asym_tensors, core_tensors = jnp.zeros((2, 0, norb, norb))
+
+    cost_func = value_and_grad(
+        partial(_compressed_cost_fn, two=two, norm_order=norm_order, prefactor=prefactor)
+    )
+
+    @jit
+    def _step(params, opt_state, asym_tensors, core_tensors):
```

Comment:
Is there a chance that the optimization might fail here, do we want to add a check in case that is a possibility.

Reply:
I am leaving the freedom to check the convergence (to whatever they may require) to the users. From my side, I allow them to adjust the optimization (and the convergence) with the `compression_kwargs`.